src/app/content/elements/popUp/FiltersList.tsx

```diff
@@ -98,25 +96,31 @@ export const FiltersListChapter = (props: FiltersListChapterProps) => (
 
 interface FiltersListProps {
   className?: string;
+  locationFilters: HighlightLocationFilters;
+  selectedLocationFilters: Set<string>;
+  selectedColorFilters?: Set<HighlightColorEnum>;
+  setFilters: (filters: Partial<SummaryFilters>) => void;
 }
 
 // tslint:disable-next-line: variable-name
-const FiltersList = ({className}: FiltersListProps) => {
-  const locationFilters = useSelector(highlightLocationFilters);
-  const selectedColorFilters = useSelector(summaryColorFilters);
-  const selectedLocationFilters = useSelector(summaryLocationFilters);
-  const dispatch = useDispatch();
+const FiltersList = ({
+  className,
+  locationFilters,
+  selectedColorFilters,
+  selectedLocationFilters,
+  setFilters,
+}: FiltersListProps) => {
 
   const onRemoveChapter = (locationId: string) => {
-    dispatch(setSummaryFilters({
+    setFilters({
       locationIds: [...selectedLocationFilters].filter(not(match(locationId))),
-    }));
+    });
   };
 
   const onRemoveColor = (color: HighlightColorEnum) => {
-    dispatch(setSummaryFilters({
-      colors: [...selectedColorFilters].filter(not(match(color))),
-    }));
+    setFilters({
+      colors: selectedColorFilters && [...selectedColorFilters].filter(not(match(color))),
+    });
```

Comment:
it may, the type is set to partial so that might allow that. we should probably change it to 
```
 [...(selectedColorFilters || [])].filter(not(match(color)))
```

so that if it is undefined we get the empty array

Reply:
the type is set to partial so i think that would be allowed. in this case that would mean we're trying to remove a filter that isn't set so i think it'd work out

Reply:
The only thing that worries me about it is that state's typing doesn't reflect it. This makes it perfectly fine to send an action
`setSummaryFilters({locationIds: undefined})`, which will make `summary.filters.locationIds` `undefined` even thought its typed as array.

Reply:
Actually, highlights' `setSummaryFilters` has the same issue.